core/src/main/java/com/sequenceiq/distrox/v1/distrox/StackOperations.java

```diff
@@ -145,139 +146,140 @@ public StackV4Response get(@NotNull StackAccessDto stackAccessDto, Long workspac
         return stackResponse;
     }
 
-    public StackViewV4Response getForInternalCrn(@NotNull StackAccessDto stackAccessDto, StackType stackType) {
+    public StackViewV4Response getForInternalCrn(ResourceAccessDto resourceAccessDto, StackType stackType) {
         LOGGER.info("Validate stack against internal user.");
-        validateAccessDto(stackAccessDto);
-        StackApiView stackApiView = stackApiViewService.retrieveStackByCrnAndType(stackAccessDto.getCrn(), stackType);
-        LOGGER.info("Query Stack (view) successfully finished with crn {}", stackAccessDto.getCrn());
+        ResourceAccessDto.validate(resourceAccessDto);
+        StackApiView stackApiView = stackApiViewService.retrieveStackByCrnAndType(resourceAccessDto.getCrn(), stackType);
+        LOGGER.info("Query Stack (view) successfully finished with crn {}", resourceAccessDto.getCrn());
         StackViewV4Response stackViewV4Response = converterUtil.convert(stackApiView, StackViewV4Response.class);
         LOGGER.info("Adding environment name to the response.");
         environmentServiceDecorator.prepareEnvironment(stackViewV4Response);
         return stackViewV4Response;
     }
 
-    public void deleteInstance(@NotNull StackAccessDto stackAccessDto, Long workspaceId, boolean forced, String instanceId) {
-        if (isNotEmpty(stackAccessDto.getName())) {
-            stackCommonService.deleteInstanceByNameInWorkspace(stackAccessDto.getName(), workspaceId, instanceId, forced);
+    public void deleteInstance(@NotNull ResourceAccessDto resourceAccessDto, Long workspaceId, boolean forced, String instanceId) {
+        if (isNotEmpty(resourceAccessDto.getName())) {
+            stackCommonService.deleteInstanceByNameInWorkspace(resourceAccessDto.getName(), workspaceId, instanceId, forced);
         } else {
-            stackCommonService.deleteInstanceByCrnInWorkspace(stackAccessDto.getCrn(), workspaceId, instanceId, forced);
+            stackCommonService.deleteInstanceByCrnInWorkspace(resourceAccessDto.getCrn(), workspaceId, instanceId, forced);
         }
     }
 
-    public void deleteInstances(StackAccessDto stackAccessDto, Long workspaceId, List<String> instanceIds, boolean forced) {
-        if (isNotEmpty(stackAccessDto.getName())) {
-            stackCommonService.deleteMultipleInstancesByNameInWorkspace(stackAccessDto.getName(), workspaceId, instanceIds, forced);
+    public void deleteInstances(ResourceAccessDto resourceAccessDto, Long workspaceId, List<String> instanceIds, boolean forced) {
+        if (isNotEmpty(resourceAccessDto.getName())) {
```

Comment:
I would refactor this and the stackCommonService.deleteMultipleInstances... method: I would pass both name and crn and would merge the byName and byCrn methods, they just differ in one line.

Also, I would do this for the other methods like putStop, putStart, etc. Those are even larger methods, differing also in just 1 line.

Reply:
this is out of scope for this pr. if it really hurts you please raise a jira for this refactor and we'll make it in the future

Reply:
I do not quite agree: in this way refactoring will be a second class citizen.